sw/device/lib/testing/test_status.c

```diff
@@ -10,26 +10,30 @@
 #include "sw/device/lib/runtime/log.h"
 
 void test_status_set(test_status_t test_status) {
-  if (kDeviceTestStatusAddress != 0) {
-    mmio_region_t sw_dv_test_status_addr =
-        mmio_region_from_addr(kDeviceTestStatusAddress);
-    mmio_region_write32(sw_dv_test_status_addr, 0x0, (uint32_t)test_status);
-  }
-
   switch (test_status) {
     case kTestStatusPassed: {
       LOG_INFO("PASS!");
+      test_status_device_write(test_status);
       abort();
       break;
     }
     case kTestStatusFailed: {
       LOG_INFO("FAIL!");
+      test_status_device_write(test_status);
       abort();
       break;
     }
     default: {
-      // Do nothing.
+      test_status_device_write(test_status);
       break;
     }
   }
 }
+
+void test_status_device_write(test_status_t test_status) {
+  if (kDeviceTestStatusAddress != 0) {
+    mmio_region_t device_test_status_addr =
+        mmio_region_from_addr(kDeviceTestStatusAddress);
+    mmio_region_write32(device_test_status_addr, 0x0, (uint32_t)test_status);
+  }
+}
```

Comment:
This should probably be a static function, as it's not used anywhere but this compilation unit.

Reply:
Fixed, thanks.